### DISCUSSION TITLE
PactSwift Example of a 'multipart/form-data;boundary=$someBoundary'

### DISCUSSION THREAD

**Discussion by joleary1987** [surpher/PactSwift#118] [General]
Hello there,

I've been looking for an example of how to use PactSwift for a request that uses `multipart/form-data` for the request body and have come up short.

Is there an example of how to write a PactSwift test for a `multipart/form-data;boundary=` request?

Thank you!
Jeff

**Comment by surpher** [comment]
Hi Jeff. Don't have anything that would share a file (yet). We might need to create a new interface for this in `PactSwift` and `PactSwiftMockServer`

What do you think? Something like the following to prepare the Request part of the interaction:
```swift
extension Interaction { 
   func withFileUpload(_ partName: String, fileName: String, contentType: String, data: Data) -> Self {
    // ...
  }
}
```

Would need to extend it here somewhere -> https://github.com/surpher/PactSwift/blob/cff1761e0fd42a6f466c03c9ac70727474856657/Sources/Model/Interaction.swift#L186
